### ISSUE TITLE
Pods unable to communicate with each other

### ISSUE THREAD

**Issue by zcatbear** [openshift/origin#19044]
I can't connect to pods on different nodes. It's first noticed when trying to launch a new build, getting `No Route to Host` for `docker-registry.default.svc` after the build was successful, but it still fails because it can't push the new image into the registry.  
I had seen this error in a previous cluster. My response was to build an entirely new cluster as that one had some serious problems, yet they still persisted.
Example:
```
error: build error: Failed to push image: After retrying 6 times, Push image still failed due to error: Get https://docker-registry.default.svc:5000/v1/_ping:  dial tcp 172.30.155.136:5000: getsockopt: no route to host
```

After this I have gone into the terminal of various pods and tried to ping other pods. I always received `no route to host` except when the two pods were on the same node.  

I have worked through the entirety of [troubleshooting the SDN](https://access.redhat.com/documentation/en-us/openshift_enterprise/3.1/html/cluster_administration/admin-guide-sdn-troubleshooting) and still can't find anything amiss, except that I can't route between the pods, but I'm not strong on Software defined networking. I have also ran the [debug.sh script](https://raw.githubusercontent.com/openshift/openshift-sdn/master/hack/debug.sh) mentioned at the end of the troubleshooting guide, after modifying it to work on boxes without root users. It didn't tell me much except that I was unable to ping other pods within the cluster (it didn't find that I could ping pods on the same node), but I was able to ping *outside* the cluster, so `ping redhat.com` worked just fine. 
##### Version
```
oc v3.7.1+ab0f056
kubernetes v1.7.6+a08f5eeb62
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://openshiftv2.isc.local:8443
openshift v3.7.0+7ed6862
kubernetes v1.7.6+a08f5eeb62
```

##### Steps To Reproduce
1. Build Cluster accepting _most_ of the defaults. I didn't change my either my `cluster_network_cidr` or my `openshift_portal_net` and left them at their default `10.128.0.0/14` and`172.30.0.0/30` respectively
2. Try to build anything. 

##### Current Result
Routing between two pods is broken, regardless of using the service network, or the pod's direct IP address, with the singular exception of using the pod's IP address when it lives on the same node the connection is coming from.
##### Expected Result
The ability to ping, or otherwise acknowledge that another pod is alive, from within the cluster, by both the 10.128.0.0/14 or the 172.30.0.0/16 networks. 

##### Additional Information
`oc adm diagnostics` didn't give me any useful output. It just complained that my master nodes were unschedulable. I join them to the cluster to make sure they're still part of the SDN.

**Comment by jwforres** [comment]
@openshift/sig-networking

**Comment by zcatbear** [comment]
Update on this:
Cluster was destroyed and rebuilt with the same problems but we were able to narrow it down further. 

### Ansible file
```
[OSEv3:children]
masters
nodes
etcd
lb
nfs
new_nodes
new_masters
new_etcd
[OSEv3:vars]
ansible_ssh_user=vagrant
ansible_become=yes
debug_level=2
openshift_deployment_type=origin
openshift_release=v3.7
openshift_disable_check=package_version
openshift_install_examples=true
openshift_docker_disable_push_dockerhub=True
osm_use_cockpit=true
osm_cockpit_plugins=['cockpit-kubernetes']
openshift_master_cluster_method=native
openshift_master_cluster_hostname=master-lb-1-openshift-test.isc.local
openshift_master_cluster_public_hostname=master-lb-1-openshift-test.isc.local
openshift_master_default_subdomain=dev-v2.isc.local
openshift_hosted_registry_storage_kind=nfs
openshift_hosted_registry_storage_access_modes=['ReadWriteMany']
openshift_hosted_registry_storage_nfs_directory=/exports/registry
openshift_hosted_registry_storage_nfs_options='*(rw,root_squash)'
openshift_hosted_registry_storage_volume_name=registry
openshift_hosted_registry_storage_volume_size=5Gi
openshift_logging_install_logging=true
openshift_logging_storage_kind=nfs
openshift_logging_storage_access_modes=['ReadWriteOnce']
openshift_logging_storage_nfs_directory=/exports/logging
openshift_logging_storage_nfs_options='*(rw,root_squash)'
openshift_logging_storage_volume_name=logging
openshift_logging_storage_volume_size=5Gi
openshift_logging_storage_labels={'storage': 'logging'}
os_sdn_network_plugin_name='redhat/openshift-ovs-multitenant'
osm_cluster_network_cidr=172.20.0.0/16
openshift_portal_net=172.21.0.0/16
openshift_set_node_ip=True
logrotate_scripts=[{"name": "syslog", "path": "/var/log/cron\n/var/log/maillog\n/var/log/messages\n/var/log/secure\n/var/log/spooler\n", "options": ["daily", "rotate 7", "compress", "sharedscripts", "missingok"], "scripts": {"postrotate": "/bin/kill -HUP `cat /var/run/syslogd.pid 2> /dev/null` 2> /dev/null || true"}}]
openshift_enable_service_catalog=true
template_service_broker_install=false
openshift_management_storage_class=nfs
[masters]
master-[1:3]-openshift-test.isc.local
[nfs]
infra-1-openshift-test.isc.local
[etcd]
master-[1:3]-openshift-test.isc.local
[lb]
master-lb-1-openshift-test.isc.local
[nodes]
master-[1:3]-openshift-test.isc.local openshift_schedulable=False
infra-1-openshift-test.isc.local openshift_node_labels="{'region': 'infra'}"
node-h-1-openshift-test.isc.local
node-l-1-openshift-test.isc.local
[new_nodes]
[new_masters]
[new_etcd]
```

### Notable changes: 
- The service network and portal network were changed to 172.20/16 and 172.21/16 respectively. 
- 3 masters were created due to an etcd issue on quorum. 
- Added a load balancer because of the 3 masters

### More discoveries
After rebuilding the cluster we were still having the same issues of not being able to push to the registry due to this error: `error: build error: Failed to push image: After retrying 6 times, Push image still failed due to error: Get https://docker-registry.default.svc:5000/v1/_ping:  dial tcp 172.21.105.56:5000: getsockopt: no route to host` . 
I got into the router pod from the namespace: default (runs on the same node as the registry) and tried to ping that address, and was successful, so once again I was able to hit pods on the same node from another node. 

I hadn't previously considered there was an issue with node communication over the ovs network but decided to try it out. I ssh'd into one of my nodes `node-h-1-openshift-test.isc.local`(heavy) and did a ping to `node-l-1-openshift-test.isc.local`(light) via `light`'s tun0 IP address and all routing was fine, same for vice versa. Neither one however, was able to hit the `infra-1-openshift-test.isc.local` (infra) and vice versa. I pulled down two openshift-compatible images into another project in the cluster and pinned them on different nodes (one light one on heavy). I was able to curl the webpage that one was hosting from the other, however from the router node I still wasn't able to. I removed the `nodeSelector` label from the registry and the router and was able to ping the registry again since they moved from the infrastructure nodes. 

### TL;DR
Node to node communication between infrastructure nodes and anything else is what is busted, I just need to know how to fix/debug it further. I have gone through the entirety of the sdn-troubleshooting guide but nothing in particular jumps out at me.

**Comment by arocki7** [comment]
I happened to have similar issues and It was finally AWS Security Group changes. After allowing the necessary ports, it worked like charm.
